pkg/api/register.go

```diff
@@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) {
 func (obj *Pod) GetObjectMeta() meta.Object                                  { return &obj.ObjectMeta }
 func (obj *Pod) GetObjectKind() unversioned.ObjectKind                       { return &obj.TypeMeta }
 func (obj *PodList) GetObjectKind() unversioned.ObjectKind                   { return &obj.TypeMeta }
+func (obj *PodList) GetListMeta() meta.List                                  { return &obj.ListMeta }
```

Comment:
It has to be a pointer so it can mutate - if you implemented on a
TypeMeta it would not have access to a pointer to type meta and your
Set* calls would silently be lost.